src/storage/src/table/state_table.rs

```diff
@@ -113,171 +116,109 @@ impl<S: StateStore> StateTable<S> {
         Ok(())
     }
 
-    pub async fn iter(&self, epoch: u64) -> StorageResult<StateTableRowIter<'_, S>> {
-        let mem_table_iter = self.mem_table.buffer.iter().peekable();
-        StateTableRowIter::new(
+    pub async fn iter(
+        &self,
+        epoch: u64,
+    ) -> StorageResult<Pin<Box<impl Stream<Item = StorageResult<Option<Row>>> + Send + '_>>> {
```

Comment:
The return type can be simplified as `BoxedStream` ?

Reply:
On second thought, if we use `BoxedStream`, then the return type seems to be `Pin<Box<dyn Stream>>` instead of `Pin<Box<impl Stream>>`. I guess it would be dynamic dispatch that makes performance a bit worse?

Reply:
Yes, I think the current way is good.

Reply:
... or we can create a trait alias for `impl Stream<Item = StorageResult<Option<Row>>`, to simplify the code here.

Reply:
Should we change all `BoxedStream` to impl style then?

Reply:
Nope. In some places, BoxedStream is inevitable (e.g. the stream executors) as they might be recursive. Also using the `impl Stream` way might impact compile speed, as the state machine could be large, so we should carefully select where we really need them.